hashedixsearch/search.py

```diff
@@ -37,21 +37,39 @@ def analyze_token(self, token):
             yield token
 
 
-def tokenize(doc, stopwords=None, ngrams=None, stemmer=None, analyzer=None):
+def tokenize(
+    doc,
+    stopwords=None,
+    ngrams=None,
+    stemmer=None,
+    analyzer=None,
+    retain_casing=False,
+    retain_punctuation=False,
+    tokenize_whitespace=False,
+):
     stopwords = stopwords or []
     stemmer = stemmer or NullStemmer()
     analyzer = analyzer or WhitespaceTokenAnalyzer()
 
     words = list(analyzer.process(doc))
     word_count = len(words)
-    doc = " ".join(words)
+    doc = ("" if tokenize_whitespace else " ").join(words)
```

Comment:
Would it be possible to fix `tokenize_whitespace=True` and remove this conditional behaviour?